### DISCUSSION TITLE
PyAEDT method/s for copying over objects!

### DISCUSSION THREAD

**Discussion by Abhi-SimuTech** [ansys/pyaedt#3818] [Q&A] [answered]
Hi,

I wish to copy all objects (solids, sheets including RLC boundaries, and ports) from one HFSS design of one project to a design (active design) of another project (active project). Or, alternatively, from other designs of the same project to the active design.

What will be the suitable PyAEDT method or sequence of methods/steps to achieve this?

Thank you in advance.

Regards
Abhi

**Comment by Samuelopez-ansys** [comment, marked as answer]
@Abhi-SimuTech 

If you want to copy a design into a new project, you can use this method:

https://aedt.docs.pyansys.com/version/stable/API/_autosummary/pyaedt.hfss.Hfss.copy_design_from.html

Or only the objects:
https://aedt.docs.pyansys.com/version/stable/API/_autosummary/pyaedt.hfss.Hfss.copy_solid_bodies_from.html

If you want something more advanced, you will need to loop all objects, PyAEDT has all the methods needed to tackle this Automation.

Thanks

**Reply by Abhi-SimuTech** [reply]
Thank you @Samuelopez-ansys 

I am actually using these methods currently.

I was wondering if there is something more straightforward available "to copy all objects (including RLC boundaries) from a design of another project to the active design of the active project" . 

But, if not, I should be able to use these to serve my purpose along with loop commands as you mentioned.

Regards

**Reply by Samuelopez-ansys** [reply]
@Abhi-SimuTech 

I think you know, but just in case, when you copy a solid, if you have this options checked in AEDT, it will copy boundaries and ports:
<img width="508" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/6a6d47c2-c067-490e-892a-f3739553406d">


Then you just need to reload the Hfss pyaedt object, and you will have everything.

To copy all objects you can use:

https://aedt.docs.pyansys.com/version/stable/API/_autosummary/pyaedt.hfss.Hfss.copy_solid_bodies_from.html

**Reply by Abhi-SimuTech** [reply]
Getting the following error, not sure this is the right way to pass names of the designs to the method?

![Capture1](https://github.com/ansys/pyaedt/assets/145813888/03f7c7a4-828e-4ed1-8c9e-b36b90f364ce)

![Capture2](https://github.com/ansys/pyaedt/assets/145813888/6a529b82-6c52-4869-9695-b779865f3049)

**Reply by Samuelopez-ansys** [reply]
@Abhi-SimuTech 

If you go to the documentation, you can will that the first argument is the PyAEDT object of the design, NOT the name.

For example:

app = Hfss(design1)

Then to copy the design1, you will need to pass app as the first argument.

**Reply by Abhi-SimuTech** [reply]
Could you provide another example, it is not clear and I am really struggling with defining the design argument.

**Reply by Abhi-SimuTech** [reply]
Basically, I wish to copy all objects from Base_Model1, Base_Model2, and Base_Model3 to Base_Model (whic is the active design) inside the project Kymeta_Base_Model.

![Capture2](https://github.com/ansys/pyaedt/assets/145813888/348e745f-0fb0-49e4-9412-9ec6a925ebc4)

**Reply by Abhi-SimuTech** [reply]
It is not clear from this example what it needs as the "design" argument. How do I modify for my above purpose?

![image](https://github.com/ansys/pyaedt/assets/145813888/4d37b14a-931b-42c8-bd68-e10319c34bc1)


![image](https://github.com/ansys/pyaedt/assets/145813888/4966e985-dc46-458b-a2aa-9ba196d8f355)

**Reply by Samuelopez-ansys** [reply]
primary_design = pyaedt.Hfss(designname="Base_Model")

second_design = pyaedt.Hfss(designname="Base_Model1")

primary_design.copy_solid_bodies_from(design=second_design)

**Reply by Abhi-SimuTech** [reply]
Thanks Samuel, it worked. Additionally, I needed to change the argument "object_list=None" from None to the list of objects from the design.

Thank you very much.

Regards
